src/Storages/System/StorageSystemServerSettings.cpp

```diff
@@ -21,15 +33,40 @@ NamesAndTypesList StorageSystemServerSettings::getNamesAndTypes()
 
 void StorageSystemServerSettings::fillData(MutableColumns & res_columns, ContextPtr context, const SelectQueryInfo &) const
 {
+    // Server settings that have been reloaded from the config file.
+    std::unordered_map<std::string, std::string> updated = {
+        {"max_server_memory_usage", std::to_string(total_memory_tracker.getHardLimit())},
+        {"allow_use_jemalloc_memory", std::to_string(total_memory_tracker.getAllowUseJemallocMmemory())},
+
+        {"max_table_size_to_drop", std::to_string(context->getMaxTableSizeToDrop())},
+        {"max_partition_size_to_drop", std::to_string(context->getMaxPartitionSizeToDrop())},
+
+        {"max_concurrent_queries", std::to_string(context->getProcessList().getMaxSize())},
+        {"max_concurrent_insert_queries", std::to_string(context->getProcessList().getMaxInsertQueriesAmount())},
+        {"max_concurrent_select_queries", std::to_string(context->getProcessList().getMaxSelectQueriesAmount())},
+
+        {"background_pool_size", std::to_string(context->getMergeMutateExecutor()->getMaxThreads())},
+        {"background_move_pool_size", std::to_string(context->getMovesExecutor()->getMaxThreads())},
+        {"background_fetches_pool_size", std::to_string(context->getFetchesExecutor()->getMaxThreads())},
+        {"background_common_pool_size", std::to_string(context->getCommonExecutor()->getMaxThreads())},
+
+        {"background_buffer_flush_schedule_pool_size", std::to_string(CurrentMetrics::get(CurrentMetrics::BackgroundBufferFlushSchedulePoolSize))},
+        {"background_schedule_pool_size", std::to_string(CurrentMetrics::get(CurrentMetrics::BackgroundSchedulePoolSize))},
+        {"background_message_broker_schedule_pool_size", std::to_string(CurrentMetrics::get(CurrentMetrics::BackgroundMessageBrokerSchedulePoolSize))},
+        {"background_distributed_schedule_pool_size", std::to_string(CurrentMetrics::get(CurrentMetrics::BackgroundDistributedSchedulePoolSize))}
+    };
+
     const auto & config = context->getConfigRef();
```

Comment:
The problem is that, for some reason, `context->getConfigRef()` returns an old version of the config after reloading. If we can fix this, `system.server_settings` table will return reloaded values automatically with no additional code required.

Reply:
The reason that I added these additinal code is that some configurations do not take effect even after reloading.
Look at the [main_config_reloader](https://github.com/ClickHouse/ClickHouse/blob/8a1363bcf1124498fd46f68e2aa671e5175bf321/programs/server/Server.cpp#L1169C8-L1169C8), we should not display the configurations that do not work.

Reply:
ok, you are right. Can you also add a new column `is_hot_reloadable`, which will be `true` for all configs in your list?

Reply:
Done.